### ISSUE TITLE
"TypeError: Cannot read property 'end' of null" when using input

### ISSUE THREAD

**Issue by AxelTerizaki** [sindresorhus/execa#384]
I'm using the `input` option to supply a string to a windows process' input like this :

```JS
const dump = await asyncReadFile(resolve(state.appPath, 'karaokemugen.sql'), 'utf-8');
await execa(binPath, options, {
			cwd: resolve(state.appPath, state.binPath.postgres),
			input: dump,
			windowsVerbatimArguments: state.os === 'win32'
		});
```

`dump` contains a string.

When trying to run this, I get a typeerror in `lib/stream.js` at line 17, and indeed, when inspecting the `spawned` ChildProcess object, I see that `stdin` is `null`. Is there something I'm doing wrong ? The docs say that you can put a string into `input` but I think the issue lies elsewhere...

I was about to turn my string into a stream but I don't think that'll solve the issue since `stdin` is `null`.

Anyone can help ? :)

EDIT: Forgot to mention I got to work around my specific case by providing the file I wanted to use as the `-f` option of psql so I didn't have to use the `input` option of execa, but I found this error a bit weird and wanted to know what I did wrong.

**Comment by ehmicky** [comment]
Hi @AxelTerizaki,

Are you sure `childProcess.stdin` is `null` and not `undefined`? There's [a bug](https://github.com/nodejs/node/issues/26852) in Node.js 10 and 11 (solved in Node 12) that makes `childProcess.stdin` `undefined` when the binary is non-executable. However the latest version of Execa fixed this.

Could you please indicate:
  - Execa version
  - Node.js version
  - OS

Thanks!

**Comment by AxelTerizaki** [comment]
Hello, and thanks for your reply :)

I'm using : 

- Node 12.4.0
- Windows 7 (did not test this on my Win10 machines)
- Execa is version 3.2.0

I'm pretty sure it's null since I did a console.log of the childprocess object in lib/stream.js to make sure the object was what I thought it was.

As for windowsVerbatimArgument it's from when I used earlier versions of execa (it has elvolved a lot lately, thanks to you.) I had trouble with some particular arguments which weren't seen by the spawned process unless I used that option. I'll try removing it later :)

**Comment by ehmicky** [comment]
Thanks for the information. Could you try the following:

```js
const {spawn} = require('child_process')

const dump = await asyncReadFile(resolve(state.appPath, 'karaokemugen.sql'), 'utf-8');
const childProcess = spawn(binPath, options, {
  cwd: resolve(state.appPath, state.binPath.postgres),
  windowsVerbatimArguments: state.os === 'win32'
});
console.log(childProcess)
```

And show what is printed? Thanks.

**Comment by ehmicky** [comment]
Based on this information, I am unfortunately not sure what caused your original error.
The next step would be: are you able to post some reproducible steps so I can debug this issue? Thanks!

**Comment by AxelTerizaki** [comment]
I tried something simple like this : 

```JS
const fs = require('fs');
const execa = require('execa');

const dump = fs.readFileSync('karaokemugen.sql', 'utf-8');
execa.sync('more', null, { stdout: 'inherit', input: dump });
```

But didn't get any errors, 

However, with this example : 

```JS
import {asyncReadFile} from './src/lib/utils/files';
import execa from 'execa';

(async () => {
	const dump = await asyncReadFile('test.ts', 'utf-8');
	console.log(typeof dump);
	const {stdout} = await execa('more', {stdio: 'inherit', input: dump});
	console.log(stdout);
})();
``` 

I'm able to reproduce the issue. `typeof dump` returns string, so it should be alright.

`asyncReadFile` is just a wrapper to promisify `fs.readFile`.

If I remove `stdio: 'inherit'` it works again, so I suppose that when using inherit, input expects a stream. (or only specify `stdout: 'inherit'`). Maybe it should be more clear in the docs, I guess.

I guess my issue is there. I might have forget to paste the correct snippet of code when opening the issue, I apologize.

I did try a few other things and was able to reproduce the issue and correctly use the `input` option. 

Sorry about taking your time for this :)

The issue can be closed, since it was after all me using incompatible options.
